### PR TITLE
Rename users and groups in documentation according to the new Wazuh naming standard

### PR DESCRIPTION
**Pull request by alromeros**
|Related issue|
|---|
|#3522|

## Description

Hi team,

This PR aims to rename to **wazuh** all the references to the old ossec-related group and users inside our documentation, which include:
- The `ossecm` user for the **wazuh-maild** daemon.
- The `ossecr` user for the **wazuh-remoted** daemon.
- Both `ossec` user and group for general purposes.

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns.

### CODE REVIEW

File: source/upgrade-guide/legacy/upgrading-elastic-stack/from-6.x-to-6.8.rst [hunk 1]
```diff
@@ -168,7 +168,7 @@ Upgrading Logstash
 
           # cp /etc/logstash/conf.d/01-wazuh.conf /backup_directory/01-wazuh.conf.bak
           # curl -so /etc/logstash/conf.d/01-wazuh.conf https://raw.githubusercontent.com/wazuh/wazuh/v|WAZUH_LATEST|/extensions/logstash/6.x/01-wazuh-local.conf
-          # usermod -a -G ossec logstash
+          # usermod -a -G wazuh logstash
```
Comment (chemamartinez) [inline source/upgrade-guide/legacy/upgrading-elastic-stack/from-6.x-to-6.8.rst:171]: I'd revert this change since this guide is to upgrade Elastic from 6.x to 6.8. Those versions don't support Wazuh v4.2 so people following this guide have Wazuh versions with the `ossec` user.

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Review by chemamartinez — APPROVED** [review]

**Review by davidfmiranda — APPROVED** [review]
LGTM
